pkg/agent/cniserver/interface_configuration_windows.go

```diff
@@ -175,7 +185,9 @@ func attachContainerLink(ep *hcsshim.HNSEndpoint, containerID, sandbox, containe
 		klog.V(2).Infof("HNS Endpoint %s already attached on container %s", ep.Id, containerID)
 	} else {
 		if err := hcsshim.HotAttachEndpoint(containerID, ep.Id); err != nil {
-			return nil, err
+			if ipam.IsInfraContainer(sandbox) || hcsshim.ErrComputeSystemDoesNotExist != err {
```

Comment:
Could non infra container reach here? I thought it's read-only?

Reply:
Yes, non infra container will reach here. We need to call `hcsshim.HotAttachEndpoint` for workload container.